modules/system-disaster-recovery/src/test/java/org/apache/ignite/internal/disaster/system/SystemDisasterRecoveryManagerImplTest.java

```diff
@@ -0,0 +1,490 @@
+    private final ClusterNode node5 = new ClusterNodeImpl(randomUUID().toString(), "node5", new NetworkAddress("host", 1005));
+
+    private final CmgMessagesFactory cmgMessagesFactory = new CmgMessagesFactory();
+    private final SystemDisasterRecoveryMessagesFactory messagesFactory = new SystemDisasterRecoveryMessagesFactory();
+
+    private final ClusterState usualClusterState = cmgMessagesFactory.clusterState()
+            .cmgNodes(Set.of(thisNodeName))
+            .metaStorageNodes(Set.of(thisNodeName))
+            .version(IgniteProductVersion.CURRENT_VERSION.toString())
+            .clusterTag(randomClusterTag(cmgMessagesFactory, CLUSTER_NAME))
+            .build();
+
+    @Captor
+    private ArgumentCaptor<ResetClusterMessage> resetClusterMessageCaptor;
+
+    @Captor
+    private ArgumentCaptor<NetworkMessage> messageCaptor;
+
+    private final SuccessResponseMessage successResponseMessage = cmgMessagesFactory.successResponseMessage().build();
+
+    @BeforeEach
+    void init() {
+        vaultManager = spy(new VaultManager(new PersistentVaultService(workDir.resolve("vault"))));
+        assertThat(vaultManager.startAsync(componentContext), willCompleteSuccessfully());
+
+        lenient().when(messagingService.respond(any(ClusterNode.class), any(NetworkMessage.class), anyLong()))
+                .thenReturn(nullCompletedFuture());
+
+        manager = new SystemDisasterRecoveryManagerImpl(
+                thisNodeName,
+                topologyService,
+                messagingService,
+                vaultManager,
+                restarter,
+                clusterStateStorage
+        );
+        assertThat(manager.startAsync(componentContext), willCompleteSuccessfully());
+    }
+
+    @AfterEach
+    void cleanup() {
+        assertThat(vaultManager.stopAsync(), willCompleteSuccessfully());
+        assertThat(manager.stopAsync(), willCompleteSuccessfully());
+    }
+
+    @Test
+    void marksNodeInitialized() {
+        manager.markNodeInitialized();
+
+        VaultEntry entry = vaultManager.get(new ByteArray(NODE_INITIALIZED_VAULT_KEY));
+        assertThat(entry, is(notNullValue()));
+        assertThat(entry.value(), is(new byte[]{1}));
+    }
+
+    @Test
+    void savesClusterName() {
+        manager.saveClusterName("name");
+
+        VaultEntry entry = vaultManager.get(new ByteArray(CLUSTER_NAME_VAULT_KEY));
+        assertThat(entry, is(notNullValue()));
+        assertThat(entry.value(), is("name".getBytes(UTF_8)));
+    }
+
+    @Test
+    void resetClusterRejectsDuplicateNodeNames() {
+        ClusterResetException ex = assertWillThrow(
+                manager.resetCluster(List.of(thisNodeName, thisNodeName)),
+                ClusterResetException.class,
+                10, SECONDS
+        );
+        assertThat(ex.getMessage(), is("New CMG node consistentIds have repetitions: [node1, node1]."));
```

Comment:
What's our recommendation, do we really put dots after `]`? I don't think so. We should check guidelines, I'm not sure about the colon as well

Reply:
Are you talking about [param1=value1] fragments? This is a different case: this is just a representation of a collection. It seemed too much to wrap it in another layer of brackets. And as this is not a 'probably special case' anymore, then we probably need a period at the end, this is my reasoning :)

Reply:
Ok, I guess we will never have any consistency in our messages